File: sdk/typescript/examples/04-queue-management.ts
// 04 — Queue management
//
// Print the current queue, demonstrate inserting tracks at different positions,
// and removing entries.
//
//   bun run examples/04-queue-management.ts

import { InsertPosition } from '../src/index.js';
import { createClient, fmtTime } from './_client.js';

const client = createClient();

const queue = await client.playlist.current();
console.log(`Queue: ${queue.amount} tracks, currently at index ${queue.index}\n`);

queue.tracks.slice(0, 10).forEach((t, i) => {
  const marker = i === queue.index ? '▶' : ' ';
  console.log(`${marker} ${(i + 1).toString().padStart(3)}. ${t.title} — ${t.artist}  (${fmtTime(t.length)})`);
});
if (queue.tracks.length > 10) console.log(`  ... and ${queue.tracks.length - 10} more`);

// --- Append a track at the end of the queue (no playback change) ----------
// Replace this path with one that exists on your system before running.
const PATH_TO_INSERT = process.argv[2];

if (PATH_TO_INSERT) {
  console.log(`\nAppending: ${PATH_TO_INSERT}`);
  await client.playlist.insertTracks([PATH_TO_INSERT], InsertPosition.Last);

  const after = await client.playlist.amount();
  console.log(`→ queue now has ${after} tracks`);
} else {
  console.log('\n(Pass a file path to insert it at the end of the queue.)');
}

File: sdk/python/examples/04_queue_management.py
"""04 — Queue management.

Show the current queue, then insert tracks from the first album in the library
at the end of the queue.

    uv run python examples/04_queue_management.py
"""

from __future__ import annotations

import asyncio

from _client import create_client  # type: ignore[import-not-found]

from rockbox_sdk import InsertPosition


async def main() -> None:
    async with create_client() as client:
        queue = await client.playlist.current()
        print(f"Queue has {queue.amount} tracks (index {queue.index}):")
        for i, t in enumerate(queue.tracks[:5]):
            marker = "▶" if i == queue.index else " "
            print(f"  {marker} {i:>3}  {t.title} — {t.artist}")
        if queue.amount > 5:
            print(f"     … and {queue.amount - 5} more")

        albums = await client.library.albums()
        if albums:
            album = albums[0]
            print(f"\nAppending album: {album.title} — {album.artist}")
            await client.playlist.insert_album(album.id, InsertPosition.LAST)
            print("→ done")


if __name__ == "__main__":
    asyncio.run(main())
